cli/chaosli/cmd/create.go

```diff
@@ -527,29 +527,39 @@ func getServices() []v1beta1.NetworkDisruptionServiceSpec {
 
 		hasPorts := confirmOption("Do you want to provide the ports affected by the disruption? (In case of no port defined, all ports of the service will be affected)", "")
 
-		ports := []v1beta1.NetworkDisruptionServicePortSpec{}
-
 		if hasPorts {
-			getServicePort := func() v1beta1.NetworkDisruptionServicePortSpec {
+			ports := []v1beta1.NetworkDisruptionServicePortSpec{}
+
+			getServicePort := func() *v1beta1.NetworkDisruptionServicePortSpec {
```

Comment:
```suggestion
			getServicePort := func() v1beta1.NetworkDisruptionServicePortSpec {
```

Reply:
using a pointer to be able to detect if there is a port provided or not, I'm not sure to understand why I should not do that. Could you provide more context to your suggested change?

Reply:
TLDR: when you have a struct, using a pointer or not should be consistent among all calls (receiver methods, returned values, etc)

In depth explanation here:
* https://dd.slack.com/archives/G01PMMNMHEZ/p1680018175024769

Here, if you concern is to validate is it empty or not you can check the returned value against an empty struct (non nil).